sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -431,7 +431,8 @@
         .setNumShards(0)
         .setCodec(TypedWrite.DEFAULT_SERIALIZABLE_CODEC)
         .setMetadata(ImmutableMap.of())
-        .setWindowedWrites(false);
+        .setWindowedWrites(false)
+        .setMaxNumWritersPerBundle(WriteFiles.DEFAULT_MAX_NUM_WRITERS_PER_BUNDLE);
```

Comment:
Beam generally tries to minimize knobs and let the runners perform decisions related bundle size, dynamic splitting etc. I'm afraid exposing knobs like this though the interface of IO connectors might promote patters that will result in runners being less-efficient in the future. And once we expose a knob we cannot take it back.

Reply:
I have also come up with another approach to my issue:  #7929. So it may be unnecessary to expose this if that is the consensus. I just want to highlight that it was a huge pain to work around this limitation so I could set a higher max. I had to copy most of the FileIO class because its all private internal stuff so that I could call WriteFiles on my own with a higher max.